subsys/net/lib/coap/coap_server.c

```diff
@@ -149,7 +148,7 @@ static int coap_server_process(int sock_fd)
 		return -errno;
 	}
 
-	ret = coap_packet_parse(&request, buf, received, options, opt_num);
+	ret = coap_packet_parse(&request, buf, MIN(received, sizeof(buf)), options, opt_num);
```

Comment:
I'm wondering if we should really still attempt to parse the packet if we know that it's incomplete?

Reply:
I think so, it parses the header (and options) only, which are needed to be able to respond.

Reply:
True. We need at least the request type.